### PR TITLE
polygon harness: add second node, connect, mine

### PR DESCRIPTION
**Pull request by buck54321**
Delete `NODES_ROOT` at start of script. Add nodekey file. Fix some bor.toml settings. Prepare files of test data for compatibility testing.

### CODE REVIEW

File: dex/testing/polygon/harness.sh [hunk 1]
```diff
@@ -235,14 +303,33 @@ EOF
...
+echo "ETH block 10 hash to use in tests is ${TEST_BLOCK10_HASH}. Saving to ${NODES_ROOT}/test_block10_hash.txt"
+cat > "${NODES_ROOT}/test_block10_hash.txt" <<EOF
+${TEST_BLOCK10_HASH}
+EOF
+
+# TEST_TOKEN_CONTRACT_ADDR=$(${HARNESS_DIR}/alpha --preload ${NODES_ROOT}/harness-ctl/contractAddress.js --exec "contractAddress(\"${TEST_TOKEN_CONTRACT_HASH}\")" | sed 's/"//g')
+# echo "Test Token contract address is ${TEST_TOKEN_CONTRACT_ADDR}. Saving to ${NODES_ROOT}/test_token_contract_address.txt"
+# cat > "${NODES_ROOT}/test_token_contract_address.txt" <<EOF
+# ${TEST_TOKEN_CONTRACT_ADDR}
+# EOF
```
Comment (ukane-philemon) [inline dex/testing/polygon/harness.sh:331]: I think we can remove this if not useful yet.

**Review by ukane-philemon — APPROVED** [review]
Running well, nodes are connecting fine and blocks are being propagated to the beta node except for several RPC read errors on alpha. 

```
TRACE[06-24|19:25:30.969] Accepted RPC connection                  conn=
DEBUG[06-24|19:25:31.017] Served rpc_modules                       reqid=1   duration="63.646µs"
DEBUG[06-24|19:25:31.023] Served eth_getHeaderByNumber             reqid=2   duration="110.778µs"
DEBUG[06-24|19:25:31.028] RPC connection read error                err=EOF
TRACE[06-24|19:25:31.129] Accepted RPC connection                  conn=
```

**Comment by buck54321** [comment]
> Running well, nodes are connecting fine and blocks are being propagated to the beta node except for several RPC read errors on alpha.
> 
> ```
> TRACE[06-24|19:25:30.969] Accepted RPC connection                  conn=
> DEBUG[06-24|19:25:31.017] Served rpc_modules                       reqid=1   duration="63.646µs"
> DEBUG[06-24|19:25:31.023] Served eth_getHeaderByNumber             reqid=2   duration="110.778µs"
> DEBUG[06-24|19:25:31.028] RPC connection read error                err=EOF
> TRACE[06-24|19:25:31.129] Accepted RPC connection                  conn=
> ```

Those are debug level. I think it's just noise. They should be trace if anything.

**Comment by ukane-philemon** [comment]
> I think it's just noise. 

I guess so.

**Comment by chappjc** [comment]
> > I think it's just noise.
> 
> I guess so.

I'm almost certain.  EOF would be a normal end in many circumstances.

File: dex/testing/polygon/harness.sh [hunk 2]
```diff
@@ -2,41 +2,54 @@
 
 SESSION="polygon-harness"
 
+SOURCE_DIR=$(pwd)
```
Comment (chappjc) [inline dex/testing/polygon/harness.sh:5]: Can kill this again I think.

File: dex/testing/polygon/harness.sh [hunk 3]
```diff
@@ -154,77 +194,105 @@ devfakeauthor = true
...
+}
+# END makenode ()
+
+echo "Starting harness"
+tmux new-session -d -s "${SESSION}" "${SHELL}"
+
+echo "Creating nodes"
+makenode "alpha" "${ALPHA_HTTP_PORT}" "${ALPHA_WS_PORT}" \
+         "${ALPHA_AUTHRPC_PORT}" "${ALPHA_NODE_PORT}" "${SESSION}:1" \
+         "${ALPHA_ADDRESS_JSON_FILE_NAME}" "${ALPHA_ADDRESS_JSON}" "${ALPHS_GRPC_PORT}" \
```
Comment (chappjc) [inline dex/testing/polygon/harness.sh:276]: ALPHS_GRPC_PORT

**Review by chappjc — CHANGES_REQUESTED** [review]

File: dex/testing/polygon/harness.sh (hunk 1 above)
Comment (buck54321) [reply dex/testing/polygon/harness.sh:331]: I'm going to follow up quickly with some contracts.

Comment (ukane-philemon) [reply dex/testing/polygon/harness.sh:330]: Okay.

**Review by chappjc — APPROVED** [review]

**Pull request merged**